lib/dry/cli/parser.rb

```diff
@@ -54,13 +54,18 @@ def self.parse_required_params(command, arguments, names, parsed_options)
 
         unless all_required_params_satisfied
           parsed_required_params_values = parsed_required_params.values.compact
+          command_name = full_command_name(names)
 
-          usage = "\nUsage: \"#{full_command_name(names)} #{command.required_arguments.map(&:description_name).join(' ')}\"" # rubocop:disable Metrics/LineLength
+          if command.subcommands.empty?
+            usage = "\nUsage: \"#{full_command_name(names)} #{command.required_arguments.map(&:description_name).join(' ')}\"" # rubocop:disable Metrics/LineLength
+          else
+            usage = "\nUsage: \"#{command_name} #{command.required_arguments.map(&:description_name).join(' ')} | #{command_name} SUBCOMMAND\"" # rubocop:disable Metrics/LineLength
+          end
```

Comment:
```ruby
usage = "\nUsage: \"#{full_command_name(names)} #{command.required_arguments.map(&:description_name).join(' ')}\""

usage += " | #{command_name} SUBCOMMAND\"" if command.subcommands.any?
```

Reply:
In your example we will have two double quotes:
```
Usage: "foo root-command ROOT_COMMAND_ARGUMENT" | foo root-command SUBCOMMAND"
```
Maybe it is better to do something like this?
```ruby
usage = "\nUsage: \"#{full_command_name(names)} #{command.required_arguments.map(&:description_name).join(' ')}" 

if command.subcommands.empty?
  usage += '"'
else
  usage += " | #{command_name} SUBCOMMAND\""
end
```

Reply:
we probably can omit it at all
`Usage: foo root-command ROOT_COMMAND_ARGUMENT | foo root-command SUBCOMMAND`

Reply:
or if you want to save double quotes, we can add them at the end once:
```ruby
usage = "\nUsage: \"#{full_command_name(names)} #{command.required_arguments.map(&:description_name).join(' ')}"
usage += " | #{command_name} SUBCOMMAND" if command.subcommands.any?
usage += '"'
```